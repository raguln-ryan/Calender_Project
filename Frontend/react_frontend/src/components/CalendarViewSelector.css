.calendar-view-selector {
  display: flex;
  border-radius: 4px;
  overflow: hidden;
  border: 1px solid #ddd;
}

.view-btn {
  background-color: #f5f5f5;
  border: none;
  padding: 8px 16px;
  cursor: pointer;
  transition: background-color 0.3s;
}

.view-btn:not(:last-child) {
  border-right: 1px solid #ddd;
}

.view-btn.active {
  background-color: #2196f3;
  color: white;
}

.view-btn:hover:not(.active) {
  background-color: #e0e0e0;
}

.dark-mode .calendar-view-selector {
  border-color: #444;
}

.dark-mode .view-btn {
  background-color: #333;
  color: #f0f0f0;
}

.dark-mode .view-btn:not(:last-child) {
  border-right-color: #444;
}

.dark-mode .view-btn.active {
  background-color: #1976d2;
}

.dark-mode .view-btn:hover:not(.active) {
  background-color: #444;
}
