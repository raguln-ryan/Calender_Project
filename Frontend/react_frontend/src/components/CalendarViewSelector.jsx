import React, { useState } from "react";

const CalendarViewSelector = ({ currentView, onViewChange }) => {
  const [showDropdown, setShowDropdown] = useState(false);
  const views = ["day", "week", "month", "year"];

  return (
    <div className="calendar-view-selector">
      <button onClick={() => setShowDropdown(!showDropdown)}>
        {currentView.charAt(0).toUpperCase() + currentView.slice(1)} â–¾
      </button>
      {showDropdown && (
        <ul className="view-dropdown">
          {views.map((v) => (
            <li key={v} onClick={() => { onViewChange(v); setShowDropdown(false); }}>
              {v.charAt(0).toUpperCase() + v.slice(1)}
            </li>
          ))}
        </ul>
      )}
    </div>
  );
};

export default CalendarViewSelector;
