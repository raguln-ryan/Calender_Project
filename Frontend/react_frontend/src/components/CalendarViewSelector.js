import React from 'react';
import './CalendarViewSelector.css';

const CalendarViewSelector = ({ currentView, onViewChange }) => {
  return (
    <div className="calendar-view-selector">
      <button 
        className={`view-btn ${currentView === 'day' ? 'active' : ''}`}
        onClick={() => onViewChange('day')}
      >
        Day
      </button>
      <button 
        className={`view-btn ${currentView === 'week' ? 'active' : ''}`}
        onClick={() => onViewChange('week')}
      >
        Week
      </button>
      <button 
        className={`view-btn ${currentView === 'month' ? 'active' : ''}`}
        onClick={() => onViewChange('month')}
      >
        Month
      </button>
    </div>
  );
};

export default CalendarViewSelector;
