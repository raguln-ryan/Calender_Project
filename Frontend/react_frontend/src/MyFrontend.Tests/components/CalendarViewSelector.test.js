// src/MyFrontend.Tests/components/CalendarViewSelector.test.jsx
import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import CalendarViewSelector from "../../components/CalendarViewSelector";

describe("CalendarViewSelector Component", () => {
  const onViewChange = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("renders all view buttons with correct text", () => {
    render(<CalendarViewSelector currentView="day" onViewChange={onViewChange} />);
    expect(screen.getByText("Day")).toBeInTheDocument();
    expect(screen.getByText("Week")).toBeInTheDocument();
    expect(screen.getByText("Month")).toBeInTheDocument();
  });

  test("applies active class to the current view correctly", () => {
    const { rerender } = render(<CalendarViewSelector currentView="day" onViewChange={onViewChange} />);
    expect(screen.getByText("Day")).toHaveClass("active");
    expect(screen.getByText("Week")).not.toHaveClass("active");
    expect(screen.getByText("Month")).not.toHaveClass("active");

    rerender(<CalendarViewSelector currentView="week" onViewChange={onViewChange} />);
    expect(screen.getByText("Week")).toHaveClass("active");
    expect(screen.getByText("Day")).not.toHaveClass("active");

    rerender(<CalendarViewSelector currentView="month" onViewChange={onViewChange} />);
    expect(screen.getByText("Month")).toHaveClass("active");
  });

  test("calls onViewChange with correct view on button click", () => {
    render(<CalendarViewSelector currentView="day" onViewChange={onViewChange} />);
    
    fireEvent.click(screen.getByText("Week"));
    expect(onViewChange).toHaveBeenCalledWith("week");

    fireEvent.click(screen.getByText("Month"));
    expect(onViewChange).toHaveBeenCalledWith("month");

    fireEvent.click(screen.getByText("Day"));
    expect(onViewChange).toHaveBeenCalledWith("day");
  });

  test("handles invalid currentView gracefully", () => {
    render(<CalendarViewSelector currentView="invalid" onViewChange={onViewChange} />);
    // No button should have active class
    expect(screen.getByText("Day")).not.toHaveClass("active");
    expect(screen.getByText("Week")).not.toHaveClass("active");
    expect(screen.getByText("Month")).not.toHaveClass("active");
  });

  test("buttons have accessibility attributes (aria-pressed)", () => {
    render(<CalendarViewSelector currentView="week" onViewChange={onViewChange} />);
    expect(screen.getByText("Week")).toHaveClass("active");
    expect(screen.getByText("Day")).not.toHaveClass("active");
    expect(screen.getByText("Month")).not.toHaveClass("active");
  });

  test("matches snapshot", () => {
    const { container } = render(<CalendarViewSelector currentView="day" onViewChange={onViewChange} />);
    expect(container).toMatchSnapshot();
  });

  test("clicking same active button still triggers onViewChange", () => {
    render(<CalendarViewSelector currentView="day" onViewChange={onViewChange} />);
    fireEvent.click(screen.getByText("Day"));
    expect(onViewChange).toHaveBeenCalledWith("day");
  });

  test("switching views multiple times works correctly", () => {
    render(<CalendarViewSelector currentView="day" onViewChange={onViewChange} />);
    fireEvent.click(screen.getByText("Week"));
    fireEvent.click(screen.getByText("Month"));
    fireEvent.click(screen.getByText("Day"));

    expect(onViewChange).toHaveBeenCalledTimes(3);
    expect(onViewChange).toHaveBeenNthCalledWith(1, "week");
    expect(onViewChange).toHaveBeenNthCalledWith(2, "month");
    expect(onViewChange).toHaveBeenNthCalledWith(3, "day");
  });
});
