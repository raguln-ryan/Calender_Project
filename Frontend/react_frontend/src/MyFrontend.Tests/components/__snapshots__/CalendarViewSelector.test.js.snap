// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarViewSelector Component matches snapshot 1`] = `
<div>
  <div
    class="calendar-view-selector"
  >
    <button
      class="view-btn active"
    >
      Day
    </button>
    <button
      class="view-btn "
    >
      Week
    </button>
    <button
      class="view-btn "
    >
      Month
    </button>
  </div>
</div>
`;
